apps/src/blockly/addons/cdoFieldButton.js

```diff
@@ -1,116 +1,77 @@
 import GoogleBlockly from 'blockly/core';
 
-const CORNER_RADIUS = 3;
-const INNER_HEIGHT = 16;
-
+/**
+ * This is a customized field which the user clicks to select an option from a customized picker,
+ * for example, the location of a sprite from a grid or a sound file from a customized modal.
+ * @param value The initial value of the field.
+ * @param validator A function that is called to validate changes to the field's value.
+ * Takes in a value & returns a validated value, or null to abort a change
+ * @param onChange The function that handles the field's editor.
```

Comment:
minor: would 'onClick' be a better name for this function, in that it's invoked when the button icon is clicked? It just so happens that both consumers of this field 'change' something about this field when clicked, but just based on what happens in this class, that isn't super clear?

Edit: also, without any other context, i think i might expect this callback to fire when the value of this field changes, not when the field itself is clicked.

Reply:
Sure - that makes sense. I'll change in the blockly wrappers as well. Thanks!